### PR TITLE
refactor dogstatsd parameters

### PR DESCRIPTION
**Pull request by clamoriniere**

### CODE REVIEW

File: stable/datadog/README.md [hunk 1]
```diff
@@ -269,17 +269,20 @@ helm install --name <RELEASE_NAME> \
 | `datadog.logsConfigContainerCollectAll`  | Collect logs from all containers                                                          | `nil`                                       |
 | `datadog.logsPointerHostPath`            | Host path to store the log tailing state in                                               | `/var/lib/datadog-agent/logs`               |
 | `datadog.apm.enabled`                    | Enable tracing from the host                                                              | `false`                                       |
+| `datadog.apm.port`                       | Used to override the default agent Apm Port                                               | `8126`                                      |
```
Comment (gbbr) [inline stable/datadog/README.md]:
```suggestion
| `datadog.apm.port`                       | Used to override the default agent APM Port                                               | `8126`                                      |
```

File: stable/datadog/templates/container-trace-agent.yaml [hunk 2]
```diff
@@ -6,22 +6,18 @@
   resources:
 {{ toYaml .Values.agents.containers.traceAgent.resources | indent 4 }}
   ports:
-  - containerPort: 8126
-    hostPort: 8126
+  - containerPort: {{ default 8126 .Values.datadog.apm.port }}
```
Comment (gbbr) [inline stable/datadog/templates/container-trace-agent.yaml]: Should it not rather be in `values.yaml`?

File: stable/datadog/values.yaml [hunk 3]
```diff
@@ -184,6 +211,12 @@ datadog:
     #
     enabled: false
 
+    ## @param port - integer - optional - default: 8126
+    ## Override the trace Agent DogStatsD port.
+    ## Note: Make sure your client is sending to the same UDP port.
+    #
+    # port:
```
Comment (gbbr) [inline stable/datadog/values.yaml]:
```suggestion
    # port: 8126
```
?

Comment (clamoriniere) [reply stable/datadog/values.yaml]: it is 2 way of doing it. 
I think it makes sens to default it in the `values.yaml`, it simplify the `{{ default 8126 .Values.datadog.apm.port }}` to `{{ .Values.datadog.apm.port }}`

**Pull request merged**